### PR TITLE
feat: element type accuracy grouping

### PR DESCRIPTION
**Pull request by Klaijan**
This PR allow grouping functionality on `evaluate.py`

To test:
Run `PYTHONPATH=. pytest test_unstructured/metrics/test_evaluate.py` or call `get_mean_grouping(<doctype or connector>, <dataframe or path to tsv file>, <export directory>, "element_type")`

### CODE REVIEW

File: unstructured/ingest/evaluate.py [hunk 1]
```diff
@@ -152,8 +152,8 @@ def measure_element_type_accuracy_command(
     help="Directory to save the output evaluation metrics to. Default to \
         your/working/dir/metrics/",
 )
-def group_text_extraction_accuracy_command(grouping: str, data_input: str, export_dir: str):
-    return group_text_extraction_accuracy(grouping, data_input, export_dir)
+def get_mean_grouping_command(grouping: str, data_input: str, export_dir: str):
+    return get_mean_grouping(grouping, data_input, export_dir)
```
Comment (pawel-kmiecik) [inline unstructured/ingest/evaluate.py:156]: Shouldn't we expose `metric_strategy` as well? From what I see in `get_mean_grouping` definition, `metric_strategy` is of type `str`, not `Optional[str]`. So this call is invalid (?).

File: unstructured/metrics/evaluate.py [hunk 2]
```diff
@@ -208,18 +214,30 @@ def group_text_extraction_accuracy(
...
+        grouped_df = _format_grouping_output(grouped_acc, grouped_miss)
+        metric_strategy = "cct"
+    elif metric_strategy == "element_type":
+        grouped_df = (
+            df.groupby(grouping)
+            .agg({"element-type-accuracy": [_mean, _stdev, _count]})
+            .rename(columns={"_mean": "mean", "_stdev": "stdev", "_count": "count"})
+        )
+        grouped_df = _format_grouping_output(grouped_df)
+        metric_strategy = "element-type"
```
Comment (pawel-kmiecik) [inline unstructured/metrics/evaluate.py:237]: This looks redundant

File: unstructured/metrics/evaluate.py [hunk 3]
```diff
@@ -208,18 +214,30 @@ def group_text_extraction_accuracy(
...
-        .agg({"cct-%missing": [_mean, _stdev, "count"]})
-        .rename(columns={"_mean": "mean", "_stdev": "stdev"})
-    )
-    grouped_df = _format_grouping_output(grouped_acc, grouped_miss)
-    _write_to_file(export_dir, f"all-{grouping}-agg-cct.tsv", grouped_df)
+    if metric_strategy == "text_extraction":
+        grouped_acc = (
+            df.groupby(grouping)
+            .agg({"cct-accuracy": [_mean, _stdev, _count]})
+            .rename(columns={"_mean": "mean", "_stdev": "stdev", "_count": "count"})
```
Comment (pawel-kmiecik) [inline unstructured/metrics/evaluate.py:221]: renaming is repeated 3 times, i think we can do it just before writing the df to file.

**Review by pawel-kmiecik — CHANGES_REQUESTED** [review]
Few minors found :)

File: unstructured/ingest/evaluate.py (hunk 1 above)
Comment (Klaijan) [reply unstructured/ingest/evaluate.py:156]: Missed that! Thank you.

File: unstructured/metrics/evaluate.py (hunk 2 above)
Comment (Klaijan) [reply unstructured/metrics/evaluate.py:237]: This just changes `_` to `-` for output file name consistency.

**Pull request merged**